jwk/helper.go

```diff
@@ -38,71 +38,41 @@ import (
 	jose "gopkg.in/square/go-jose.v2"
 )
 
-func EnsureAsymmetricKeypairExists(ctx context.Context, r InternalRegistry, g KeyGenerator, set string) error {
-	_, _, err := AsymmetricKeypair(ctx, r, g, set)
+func EnsureAsymmetricKeypairExists(ctx context.Context, r InternalRegistry, alg, set string) error {
+	_, _, err := GetOrGenerateKeys(ctx, r, r.KeyManager(), set, set, alg)
 	return err
 }
 
-func AsymmetricKeypair(ctx context.Context, r InternalRegistry, g KeyGenerator, set string) (public, private *jose.JSONWebKey, err error) {
-	priv, err := GetOrCreateKey(ctx, r, g, set, "private")
-	if err != nil {
-		return nil, nil, err
-	}
-
-	pub, err := GetOrCreateKey(ctx, r, g, set, "public")
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return pub, priv, nil
-}
-
-func GetOrCreateKey(ctx context.Context, r InternalRegistry, g KeyGenerator, set, prefix string) (*jose.JSONWebKey, error) {
-	keys, err := r.KeyManager().GetKeySet(ctx, set)
+func GetOrGenerateKeys(ctx context.Context, r InternalRegistry, m Manager, set, kid, alg string) (public, private *jose.JSONWebKey, err error) {
+	keys, err := m.GetKeySet(ctx, set)
 	if errors.Is(err, x.ErrNotFound) || keys != nil && len(keys.Keys) == 0 {
 		r.Logger().Warnf("JSON Web Key Set \"%s\" does not exist yet, generating new key pair...", set)
-		keys, err = createKey(ctx, r, g, set)
+		keys, err = m.GenerateKeySet(ctx, set, kid, alg, "sig")
```

Comment:
Does this method also store the keys? If so, it might make sense to name it: `GenerateAndPersistKeySet`